runtime/queries/metricsview_search.go

```diff
@@ -0,0 +1,306 @@
+	res, ok := v.(*runtimev1.MetricsViewSearchResponse)
+	if !ok {
+		return fmt.Errorf("MetricsViewSearch: mismatched unmarshal input")
+	}
+	q.Result = res
+	return nil
+}
+
+func (q *MetricsViewSearch) Resolve(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
+	mv, lastUpdatedOn, err := lookupMetricsView(ctx, rt, instanceID, q.MetricsViewName)
+	if err != nil {
+		return err
+	}
+	resolvedSecurity, err := rt.ResolveMetricsViewSecurity(q.SecurityAttributes, instanceID, mv, lastUpdatedOn)
+	if err != nil {
+		return err
+	}
+	if resolvedSecurity != nil && !resolvedSecurity.Access {
+		return ErrForbidden
+	}
+	for _, d := range q.Dimensions {
+		if !checkFieldAccess(d, resolvedSecurity) {
+			return ErrForbidden
+		}
+	}
+
+	olap, release, err := rt.OLAP(ctx, instanceID, mv.Connector)
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	if olap.Dialect() != drivers.DialectDuckDB && olap.Dialect() != drivers.DialectDruid && olap.Dialect() != drivers.DialectClickHouse && olap.Dialect() != drivers.DialectPinot {
+		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
+	}
+
+	if mv.TimeDimension == "" && !isTimeRangeNil(q.TimeRange) {
+		return fmt.Errorf("metrics view '%s' does not have a time dimension", mv)
+	}
+
+	if !isTimeRangeNil(q.TimeRange) {
+		start, end, err := ResolveTimeRange(q.TimeRange, mv)
+		if err != nil {
+			return err
+		}
+		q.TimeRange = &runtimev1.TimeRange{
+			Start: timestamppb.New(start.In(time.UTC)),
+			End:   timestamppb.New(end.In(time.UTC)),
+		}
+	}
+
+	if olap.Dialect() == drivers.DialectDruid {
+		ok, err := q.executeSearchInDruid(ctx, rt, instanceID, mv.Table, resolvedSecurity)
+		if err != nil || ok {
+			return err
+		}
+	}
+
+	sql, args, err := q.buildSearchQuerySQL(mv, olap.Dialect(), resolvedSecurity)
+	if err != nil {
+		return err
+	}
+
+	rows, err := olap.Execute(ctx, &drivers.Statement{
+		Query:            sql,
+		Args:             args,
+		Priority:         priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return nil
+	}
+	defer rows.Close()
+
+	q.Result = &runtimev1.MetricsViewSearchResponse{Results: make([]*runtimev1.MetricsViewSearchResponse_SearchResult, 0)}
+	for rows.Next() {
+		res := map[string]any{}
+		err := rows.MapScan(res)
+		if err != nil {
+			return err
+		}
+
+		dimName, ok := res["dimension"].(string)
+		if !ok {
+			return fmt.Errorf("unknown result dimension: %q", dimName)
+		}
+
+		v, err := structpb.NewValue(res["value"])
+		if err != nil {
+			return err
+		}
+
+		q.Result.Results = append(q.Result.Results, &runtimev1.MetricsViewSearchResponse_SearchResult{
+			Dimension: dimName,
+			Value:     v,
+		})
+	}
+
+	return nil
+}
+
+func (q *MetricsViewSearch) Export(ctx context.Context, rt *runtime.Runtime, instanceID string, w io.Writer, opts *runtime.ExportOptions) error {
+	return nil
+}
+
+var druidDNSFixer = regexp.MustCompile(`/v2/sql(?:/avatica-protobuf)?/?`)
```

Comment:
Should be `DSN` not `DNS`

nit: `avatica-protobuf` should not be present after getting dsn from druid driver. Also we can call this var `druidSqlDSN`